pkg/mimir/modules.go

```diff
@@ -860,7 +861,7 @@ func (t *Mimir) initQueryFrontend() (serv services.Service, err error) {
 	// If the query-frontend is running in the same process as the query-scheduler and
 	// the query-scheduler hasn't been explicitly configured, Mimir will default to trying
 	// to connect to a scheduler on localhost on its own gRPC listening port.
-	if t.Cfg.isAnyModuleEnabled(QueryScheduler, Read, All) && !t.Cfg.Frontend.FrontendV2.IsSchedulerConfigured() {
+	if t.Cfg.isQuerySchedulerEnabled() && !t.Cfg.Frontend.FrontendV2.IsSchedulerConfigured() {
```

Comment:
Note to reviewers: this logic seemed incorrect to me: the old version was checking if the `read` target was enabled, but the query-scheduler runs in the `backend` target in read/write mode. There was further discussion of this in https://github.com/grafana/mimir/pull/12200#discussion_r2248882311.

This is also the reason for the change in `TestMimir` in `mimir_test.go`.